tests/integration_tests/test_visualisations.py

```diff
@@ -0,0 +1,1604 @@
+                    '--ground_truth_metadata',
+                    ground_truth_metadata,
+                    '-od', exp_dir_name]
+    test_cmd_png = test_cmd_pdf.copy() + ['-ff', 'png']
+    commands = [test_cmd_pdf, test_cmd_png]
+    vis_patterns = [vis_output_pattern_pdf, vis_output_pattern_png]
+
+    for command, viz_pattern in zip(commands, vis_patterns):
+        result = subprocess.run(
+            command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        figure_cnt = glob.glob(viz_pattern)
+
+        assert 0 == result.returncode
+        assert 2 == len(figure_cnt)
+
+    shutil.rmtree(exp_dir_name, ignore_errors=True)
+    shutil.rmtree('results', ignore_errors=True)
+    for file in glob.glob(experiment_source_data_name + '.*'):
+        try:
+            os.remove(file)
+        except OSError as e:  # if failed, report it back to the user
+            print("Error: %s - %s." % (e.filename, e.strerror))
+
+
+def test_visualisation_compare_performance_output_saved(csv_filename):
+    """Ensure pdf and png figures from the experiments can be saved.
+
+    Compare performance between two models. To reduce test complexity
+    one model is compared to it self.
+
+    :param csv_filename: csv fixture from tests.fixtures.filenames.csv_filename
+    :return: None
+    """
+    input_features = [text_feature(reduce_output=None, encoder='rnn')]
+    output_features = [text_feature(reduce_input=None, decoder='tagger')]
+
+    # Generate test data
+    rel_path = generate_data(input_features, output_features, csv_filename)
+    encoder = 'cnnrnn'
+    input_features[0]['encoder'] = encoder
+    exp_dir_name = run_experiment(input_features, output_features,
+                                  data_csv=rel_path)
+    vis_output_pattern_pdf = exp_dir_name + '/*.pdf'
+    vis_output_pattern_png = exp_dir_name + '/*.png'
+    experiment_source_data_name = csv_filename.split('.')[0]
+    test_stats = exp_dir_name + '/test_statistics.json'
+
+    test_cmd_pdf = ['python',
+                    '-m',
+                    'ludwig.visualize',
+                    '--visualization',
+                    'compare_performance',
+                    '--test_statistics',
+                    test_stats,
+                    test_stats,
+                    '-m',
+                    'Model1',
+                    'Model2',
+                    '-od', exp_dir_name]
+    test_cmd_png = test_cmd_pdf.copy() + ['-ff', 'png']
+
+    commands = [test_cmd_pdf, test_cmd_png]
+    vis_patterns = [vis_output_pattern_pdf, vis_output_pattern_png]
+
+    for command, viz_pattern in zip(commands, vis_patterns):
+        result = subprocess.run(
+            command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        figure_cnt = glob.glob(viz_pattern)
+
+        assert 0 == result.returncode
+        assert 2 == len(figure_cnt)
+
+    shutil.rmtree(exp_dir_name, ignore_errors=True)
```

Comment:
This line can be in the run_experiment method.

Reply:
HI guys,

I am glad you find the tests OK.

I have moved the line out of the run_experiment method on purpose in to each of the test cases. This is needed because i need to access files generated in that directory after the experiment has finished.

Each test case now cleans the files that it creates (experiment files, visualisation files).

Reply:
Got it!